### PR TITLE
Add EXP-3349 [v113] Bring Nimbus Messaging testing up to parity with Android

### PR DESCRIPTION
**Pull request by jhugman**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/EXP-3349)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/TODO)

### Description
This PR adds testing for Nimbus Messaging (né GleanPlumb), especially for triggers and messages added via the FML.

It tests that the messages are constructed:
* all messages have valid styles, triggers, etc
* all triggers are syntactically valid JEXL.

### Pull requests checks where applicable
- [ ] Fill in the three TODOs above (tickets number and description of your work)
- [x] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [x] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Review by yoanarios — COMMENTED** [review]
LGTM just some minor comments, thank you for adding unit test

File: Tests/ClientTests/NimbusMessagingTriggerTests.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    func testTriggers() throws {
+        let helper = self.helper
+        let triggers = feature.triggers
+
+        var badJexlExpressions = [String: String]()
+        triggers.forEach { (key, expression) in
+            do {
+                _ = try helper.evalJexl(expression: expression)
+            } catch {
+                print("Failed to evaluate \(key) expression: \(expression)")
```
Comment (yoanarios) [inline Tests/ClientTests/NimbusMessagingTriggerTests.swift:42]: We don't use print in the code anymore we could use XCTFail in this case
```suggestion
                XCTFail("Failed to evaluate \(key) expression: \(expression)")
```

File: Client/Experiments/Messaging/GleanPlumbMessageManager.swift [hunk 2]
```diff
@@ -201,10 +201,7 @@ class GleanPlumbMessageManager: GleanPlumbMessageManagerProtocol {
     // MARK: - Misc. Private helpers
 
     /// - Returns: All well-formed, non-expired messages for a surface in descending priority order for a specified surface.
```
Comment (thatswinnie) [inline Client/Experiments/Messaging/GleanPlumbMessageManager.swift:203]:
```suggestion
    /// - Returns: All well-formed, non-expired messages in descending priority order.
```

File: Client/Experiments/Messaging/GleanPlumbMessageManager.swift [hunk 3]
```diff
@@ -201,10 +201,7 @@ class GleanPlumbMessageManager: GleanPlumbMessageManagerProtocol {
...
-    private func getAllValidMessagesFor(
-        _ surface: MessageSurfaceId,
-        with feature: Messaging
-    ) -> [GleanPlumbMessage] {
+    func getMessages(_ feature: Messaging) -> [GleanPlumbMessage] {
         // All these are non-expired, well formed, and descending priority messages for a requested surface.
```
Comment (thatswinnie) [inline Client/Experiments/Messaging/GleanPlumbMessageManager.swift:205]:
```suggestion
        // All these are non-expired, well formed, and descending priority messages.
```

**Pull request merged**